include/openssl/ssl.h.in

```diff
@@ -1511,6 +1511,19 @@ const char *SSL_group_to_name(SSL *s, int id);
 int SSL_set0_tmp_dh_pkey(SSL *s, EVP_PKEY *dhpkey);
 int SSL_CTX_set0_tmp_dh_pkey(SSL_CTX *ctx, EVP_PKEY *dhpkey);
 
+typedef enum {
+    ssl_grease_cipher = 0,
+    ssl_grease_group,
+    ssl_grease_extension1,
+    ssl_grease_extension2,
+    ssl_grease_version,
+    ssl_grease_ticket_extension,
+    ssl_grease_last_index = ssl_grease_ticket_extension,
+} SSL_GREASE_INDEX;
+
+uint16_t SSL_get_grease_value(SSL *s,
+                              SSL_GREASE_INDEX index);
```

Comment:
This and the enum probably should not be exported. Rather the idea (I wrote the original. :-) ) was that the library would internally call into these as needed as part of serializing the ClientHello, etc.

The whole business with `grease_seed` is just because there are cases (HelloVerifyRequest, HelloRetryRequest), where variations on a ClientHello are serialized multiple times, and they need to be consistent. (Originally we even just sampled client_random, but switched to a separate seed later since that seemed tidier.)

Reply:
Ok, thanks so much for keeping an eye on this! I watched one of your talks on youtube so I could gather what your thoughts were on some of this :)

Reply:
Well, the other half is that you probably need to actually call the function in the library to go send GREASE ciphers, etc. Otherwise there's just a floating unused function in here.

Haha, I can't watch that talk. It's too embarrassing to see myself speak. :-)

Reply:
> Well, the other half is that you probably need to actually call the function in the library to go send GREASE ciphers

Yeah of course, I'm a fan of pushing code early. Nothing more disheartening that writing a load of code a certain way only for someone more knowledgeable to untangle it all :)

Expect to see frequent updates in this PR, I was on vacation the last day or so. St. Patrick's day and all that.